stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -387,14 +387,17 @@ get_value(sym::QuoteNode, fn) = isdefined(fn, sym.value) ? (getfield(fn, sym.val
 get_value(sym::GlobalRef, fn) = get_value(sym.name, sym.mod)
 get_value(sym, fn) = (sym, true)
 
-# Return the value of a getfield call expression
-function get_value_getfield(ex::Expr, fn)
-    # Example :((top(getfield))(Base,:max))
-    val, found = get_value_getfield(ex.args[2],fn) #Look up Base in Main and returns the module
-    (found && length(ex.args) >= 3) || return (nothing, false)
-    return get_value_getfield(ex.args[3], val) #Look up max in Base and returns the function if found.
-end
-get_value_getfield(sym, fn) = get_value(sym, fn)
+# Return the type of a getfield call expression
+function get_type_getfield(ex::Expr, fn::Module)
+    length(ex.args) == 3 || return Any, false # should never happen, but just for safety
+    obj, x = ex.args[2:3]
+    objt, found = get_type_getfield(obj, fn)
+    found || return Any, false
+    fld = isa(x, QuoteNode) ? x.value : x
+    isdefined(objt, fld) || return Any, false
```

Comment:
This isn't necessary
```suggestion
```

Reply:
true, I can't understand why the added test has been passed with the existence of this check ...